spec/factories/course_assessment_submissions.rb

```diff
@@ -22,10 +22,14 @@
     end
 
     trait :graded do
+      transient do
+        with_grade nil
+      end
+
       submitted
       after(:build) do |submission, evaluator|
         submission.answers.each do |answer|
-          answer.grade = Random::DEFAULT.rand(answer.question.maximum_grade)
+          answer.grade = evaluator.with_grade || Random::DEFAULT.rand(answer.question.maximum_grade)
```

Comment:
I may have misunderstood what you said in the previous comment. Can you explain what you meant by "these can be shortened to use factory attributes" for the previous comment at line 175?